src/riak_kv_pb_timeseries.erl

```diff
@@ -87,8 +88,8 @@ init() ->
 decode(Code, Bin) ->
     Msg = riak_pb_codec:decode(Code, Bin),
     case Msg of
-        #tsqueryreq{query = Q}->
-            case catch decode_query(Q) of
+        #tsqueryreq{query = Q, cover_context = Cover}->
+            case catch decode_query(Q, Cover) of
```

Comment:
Best to use a try-catch like this:
```` erlang
try decode_query(Q, Cover) of
  {ok, DecodedQuery} -> ...
catch
   T:E -> ...
end
````

Allows for better reporting of the error.